### PR TITLE
Fix CalDAV events sync

### LINKED ISSUE #1494: Caldav crash during sync
```
2022-03-30T18:58:43+0200 <error> calendar.syncUserCalendars.js:104 (Promise.map.concurrency) TypeError: Cannot read property 'childNodes' of undefined
    at /src/server/services/caldav/lib/calendar/calendar.requests.js:131:71
    at Array.map (<anonymous>)
    at CalDAVHandler.requestEventsData (/src/server/services/caldav/lib/calendar/calendar.requests.js:129:80)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Promise.map.concurrency (/src/server/services/caldav/lib/calendar/calendar.syncUserCalendars.js:102:22)
```

https://community.gladysassistant.com/t/gladys-4-service-caldav/5838/69?u=pierre-gilles

### PR DESCRIPTION
**Pull request by bertrandda**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Handle full day events without end date (equivalent to 1 day full day event)

Also, if too many changes are requested and `507 Insufficient Storage` status appears, it request again for more (and again while not finished)

Fix #1494 by handling calendar with special chars in url (`+`,`@`...)

### CODE REVIEW

File: server/services/caldav/lib/calendar/calendar.formaters.js [hunk 1]
```diff
@@ -167,11 +167,23 @@ function formatEvents(caldavEvents, gladysCalendar) {
...
       ) {
         newEvent.full_day = true;
       }
 
+      if (newEvent.full_day && !caldavEvent.end) {
+        newEvent.end = this.dayjs
+          .tz(
+            this.dayjs(caldavEvent.start)
+              .add(1, 'day')
+              .format('YYYY-MM-DDTHH:mm:ss'),
```
Comment (Pierre-Gilles) [inline server/services/caldav/lib/calendar/calendar.formaters.js:181]: Is this really the expected behavior ? In your experience, the "start" will be at 00:00 in this case ?

Just to be sure it'll work with scenes, if it's a full day event, we need the event to be from 00:00 to 00:00 (Or 23:59 )

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Thanks for this PR ! :) 

I tried syncing my Google calendar, and it doesn't work anymore (it was working fine before)

```
2022-03-18T08:59:42+0000 <error> calendar.syncUserCalendars.js:72 (Promise.map.concurrency) TypeError: Cannot read property 'props' of undefined
    at CalDAVHandler.requestChanges (/Users/pierregilles/code/gladys/server/services/caldav/lib/calendar/calendar.requests.js:88:59)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Promise.map.concurrency (/Users/pierregilles/code/gladys/server/services/caldav/lib/calendar/calendar.syncUserCalendars.js:70:26)
```

**Comment by Pierre-Gilles** [comment]
Waiting for this before merging: https://community.gladysassistant.com/t/gladys-4-service-caldav/5838/79?u=pierre-gilles

**Comment by bertrandda** [comment]
Last commit fix #1494

**Pull request merged**